dynamiqs/utils/utils.py

```diff
@@ -316,26 +337,28 @@ def ptrace(x: Tensor, keep: int | tuple[int, ...], dims: tuple[int, ...]) -> Ten
 
     # trace out x over unkept dimensions
     batch_dims = x.shape[:-2]
-    if is_ket(x):
+    if is_ket(x) or is_bra(x):
```

Comment:
Is it really 100% symmetric for a bra 🤔 I don't see why it works.

Reply:
Yes simply because the first thing we do is squeeze the state with `x = x.view(-1, *dims)`. So whether it's a ket or bra is irrelevant.